com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -368,70 +369,82 @@ public static String getLib(Platform platform, String name) throws IOException {
+        addOption(options, "k", "keep-unused", false, "Keep unused resources in archived output", true);
 
-        options.addOption(null, "exclude-build-folder", true, "Comma separated list of folders to exclude from the build");
+        addOption(options, null, "exclude-build-folder", true, "Comma separated list of folders to exclude from the build", true);
 
-        options.addOption("br", "build-report", true, "Filepath where to save a build report as JSON");
-        options.addOption("brhtml", "build-report-html", true, "Filepath where to save a build report as HTML");
+        addOption(options, "br", "build-report", true, "Filepath where to save a build report as JSON", false);
+        addOption(options, "brhtml", "build-report-html", true, "Filepath where to save a build report as HTML", false);
 
-        options.addOption(null, "build-server", true, "The build server (when using native extensions)");
-        options.addOption(null, "defoldsdk", true, "What version of the defold sdk (sha1) to use");
-        options.addOption(null, "binary-output", true, "Location where built engine binary will be placed. Default is \"<build-output>/<platform>/\"");
+        addOption(options, null, "build-server", true, "The build server (when using native extensions)", true);
+        addOption(options, null, "defoldsdk", true, "What version of the defold sdk (sha1) to use", true);
+        addOption(options, null, "binary-output", true, "Location where built engine binary will be placed. Default is \"<build-output>/<platform>/\"", true);
 
-        options.addOption(null, "use-vanilla-lua", false, "Only ships vanilla source code (i.e. no byte code)");
-        options.addOption(null, "archive-resource-padding", true, "The alignment of the resources in the game archive. Default is 4");
+        addOption(options, null, "use-vanilla-lua", false, "Only ships vanilla source code (i.e. no byte code)", true);
+        addOption(options, null, "archive-resource-padding", true, "The alignment of the resources in the game archive. Default is 4", true);
 
-        options.addOption("l", "liveupdate", true, "Yes if liveupdate content should be published");
+        addOption(options, "l", "liveupdate", true, "Yes if liveupdate content should be published", true);
 
-        options.addOption("ar", "architectures", true, "Comma separated list of architectures to include for the platform");
+        addOption(options, "ar", "architectures", true, "Comma separated list of architectures to include for the platform", false);
 
-        options.addOption(null, "settings", true, "Path to a game project settings file. More than one occurrance are allowed. The settings files are applied left to right.");
+        addOption(options, null, "settings", true, "Path to a game project settings file. More than one occurrance are allowed. The settings files are applied left to right.", true);
```

Comment:
This one is interesting.

It refers to project settings (not build options).
I don't think we really rely on the contents of the project properties (`Project.getProjectProperties()`) when building content. Except for when bundling, which we (currently) handle separately/locally. We might want to make bundling into regular tasks in the future.

Between two builds, the contents of this settings file may change, in which case it which will not trigger a fetch from the cache.
Similarly, the same file may be renamed (which I think is less severe).

Reply:
Ah, yes, you are right. It's the contents that has an impact, not the file itself.